pynautobot/core/endpoint.py

```diff
@@ -690,3 +690,39 @@ def run(self, *args, api_version=None, **kwargs):
         ).post(args[0] if args else kwargs)
 
         return response_loader(req, self.return_obj, self)
+
+
+class GraphqlEndpoint(Endpoint):
+    """Extend Endpoint class to support run method for graphql queries."""
+
+    def run(self, *args, **kwargs):
+        """Runs a saved graphql query based on the query_id provided.
+
+        Takes a kwarg of `query_id` to specify the query that should be run.
+
+        Args:
+            *args (str, optional): Used as payload for POST method
+                to the API if provided.
+            **kwargs (str, optional): Any search argument the
+                endpoint accepts can be added as a keyword arg.
+
+        Returns:
+            An API response from the execution of the saved graphql query.
+
+        Examples:
+            To run a query with the following variables:
+            >>> query = nb.extras.graphql_queries.get("Example")
+            >>> query.run(
+                    variables={"foo": "bar"})
+                )
+        """
+
+        if not kwargs.get("query_id"):
+            raise ValueError('Keyword Argument "query_id" is required to run a query.')
```

Comment:
Why not make `query_id` a required arg?

Reply:
@tsm1th I don't think you understood what I was saying. I meant doing `def run(query_id, *args, **kwargs):` and letting the standard invalid arg error from python be raised, and thus we don't need these two lines.

Reply:
I could make it a positional argument, but I would have to change the `*args` logic or remove it.  I'm currently allowing a custom payload to be passed in via args.

Making that change results in the following new error instead:
`TypeError: GraphqlEndpoint.run() missing 1 required positional argument: 'query_id'`

Reply:
This would work:

```python
class GraphqlQueries(Record):
    def run(self, *args, **kwargs):
        """Run a graphql query from a saved graphql instance."""
        return GraphqlEndpoint(self.api, self.api.extras, "graphql_queries").run(self.id, *args, **kwargs)
```

And in the endpoint:
```python
class GraphqlEndpoint(Endpoint):
    """Extend Endpoint class to support run method for graphql queries."""

    def run(self, query_id, *args, **kwargs):
```